lib/rubocop/cop/rspec/sort_tags.rb

```diff
@@ -0,0 +1,333 @@
+            end
+          end
+
+          def include_symbols?
+            if @style == :hash
+              @sym_nodes.any?
+            else
+              !symbols_sorted?
+            end
+          end
+
+          def include_hash?
+            return false unless @hash_node
+            return !hash_sorted? unless @style == :prefer_symbols
+
+            hash_pairs.any? do |pair_node|
+              pair_node.children.last.true_type?
+            end
+          end
+
+          def sorted?
+            symbols_sorted? && hash_sorted?
+          end
+
+          def symbols_sorted?
+            @sym_nodes == @sym_nodes.sort_by(&:value)
+          end
+
+          def hash_sorted?
+            return true unless @hash_node
+
+            relevant_hash_pairs == relevant_hash_pairs.sort_by do |pair_node|
+              pair_node.children.first.value.to_s
+            end
+          end
+
+          def relevant_hash_pairs
+            if @style == :prefer_symbols
+              hash_pairs.reject do |pair_node|
+                pair_node.children.last.true_type?
+              end
+            else
+              hash_pairs
+            end
+          end
+
+          def processed_source
+            @cop.processed_source
+          end
+        end
+
+        class CrimeSceneCleaner < Deputy # :nodoc:
+          def cleanup
+            new_tags = sorted_new_symbols
+            new_tags += sorted_new_pairs.map { |k, v| "#{k}: #{v}" }
+
+            new_tags.join(', ')
+          end
+
+          private
+
+          def sorted_new_symbols
+            return [] if @style == :hash
+
+            symbols = @sym_nodes.map(&:value) + new_symbols_from_hash
+
+            symbols.map(&:inspect).sort
+          end
+
+          def new_symbols_from_hash
+            return [] unless @style == :prefer_symbols && @hash_node
+
+            hash_pairs.map do |pair_node|
+              next unless pair_node.children.last.true_type?
+
+              pair_node.children.first.value
+            end.compact
+          end
+
+          def sorted_new_pairs
+            new_pairs.sort_by(&:first)
+          end
+
+          def new_pairs
+            pairs = hash_pairs.map do |pair_node|
+              [
+                pair_node.children.first.value.to_s,
+                pair_node.children.last.source
+              ]
+            end
+
+            adjust_pairs_to_style(pairs)
+          end
+
+          def adjust_pairs_to_style(pairs)
+            case @style
+            when :hash
+              @sym_nodes.each_with_object(pairs) do |sym_node, array|
+                array << [sym_node.value.to_s, 'true']
+              end
+            when :prefer_symbols
+              pairs.reject { |pair| pair.last == 'true' }
+            when :sort_only
+              pairs
+            end
+          end
+        end
+
+        private_constant :CrimeSceneCleaner, :CrimeSceneInvestigator, :Deputy
+      end
+    end
+  end
+end
```

Comment:
Even though topics of sorting and converting from symbols to hash are closely related, would you agree that considering the length of this, it would better be two separate cops?

I believe there's a metric that limits the cop's LoC to 100 lines. Somehow you've managed to trick it, perhaps Deputies are the reason. Still, it's quite large to comprehend quickly.

Reply:
I see that this cop only suggests ordering, right? Changing from `:foo` to `foo: true` and the other way around was handy, too. Do you plan to submit it as yet another cop?